### DISCUSSION TITLE
Static props on expo-router static rendering?

### DISCUSSION THREAD

**Discussion by jhuayta-rbi** [expo/expo#30302] [Archived: Questions]
Is there a way to inject other props than just page dynamic param? I want to prefect static content on `generateStaticParams` but only dynamic param is available on `useLocalSearchParams`. Also page component doesn't receive props.
There is no documentation about this static props on expo docs https://docs.expo.dev/router/reference/static-rendering/ 🤔

**Comment by ChronSyn** [comment]
Just touching on this subject again, I feel like this is a major oversight in the SSG space.

For example, I'm trying to build out a basic blog example, where my content is just MDX files that are included in the project. The idea being to have it self-contained, not having to worry about a database or similar (which I view as one of the key benefits of SSG).

`generateStaticParams` enables me to tell it what routes it should generate files for, generated from file paths for example via `readdir`, but that functionality is obviously not going to be available in the rendered content (since it's a node function for the local fs).

This means that I can't even call into `generateStaticParams` in order to pull data out, because this will in turn attempt to call the `readdir` function. The `Page` function (i.e. the default export from a `/app/` route) receives nothing in its params except the segment.

I can return static content in `generateStaticParams`, as long as it doesn't call any API's which are only available on the build-time, but then that kind of defeats the purpose of SSG.

I noticed that a PR with `useLoaderData` recently got merged (https://github.com/expo/expo/pull/39570) - which seems like it might be the closest I could find to what I'm trying to achieve: having my static content accessible via a prop or hook.

Is that PR going to actually resolve this issue? Or, can we get some solid examples on actually including static content into pages via `generateStaticParams` which can then be automatically injected into our `Page` function / exported component?

**Reply by mwood23** [reply]
Struggling with the same thing. Started a thread in Discord about it: https://discord.com/channels/695411232856997968/1462671261703471176
